master/internal/api_experiment.go

```diff
@@ -254,25 +258,49 @@ func (a *apiServer) PreviewHPSearch(
 	if err != nil {
 		return nil, status.Errorf(codes.InvalidArgument, "error parsing experiment config: %s", err)
 	}
-	config := model.DefaultExperimentConfig(&a.m.config.TaskContainerDefaults)
-	if err = json.Unmarshal(bytes, &config); err != nil {
-		return nil, status.Errorf(codes.InvalidArgument, "error parsing experiment config: %s", err)
+
+	// Parse the provided experiment config.
+	config, err := expconf.ParseAnyExperimentConfigYAML(bytes)
+	if err != nil {
+		return nil, status.Errorf(
+			codes.InvalidArgument, "invalid experiment configuration: %s", err,
+		)
+	}
+
+	// Get the useful subconfigs for preview search.
+	if config.RawSearcher == nil {
```

Comment:
q: wouldn't `ParseAnyExperimentConfigYAML` have already checked sane bytes, guaranteeing this is here?

Reply:
The searcher is not required because it can be provided by templates.

Reply:
didn't realize this